template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs

```diff
@@ -113,15 +116,17 @@ public class InputModel
                 return Page();
             }
 
-            var user = await _userManager.GetUserAsync(HttpContext.User);
+            var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
                 return RedirectToPage("/Error");
             }
 
             var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
-            var callbackUrl = Url.EmailConfirmationLink(user.Id, code, Request.Scheme);
-            await _emailSender.SendEmailConfirmationAsync(Input.Email, callbackUrl);
+            var userId = await _userManager.GetUserIdAsync(user);
```

Comment:
I assume the user ID never changes, so I think calling `_userManager.GetUserId(User)` should be fine. For the username, I think it really depends if you are ok with username stored in the cookie, or if you want the most up to date username from the identity store. @HaoK Does that sound right?